oak_functions/client/rust/src/rekor.rs

```diff
@@ -0,0 +1,181 @@
+/// client, it is omitted from this struct.
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+pub struct LogEntryVerification {
+    // Base64-encoded signature over the body, integratedTime, logID, and logIndex.
+    #[serde(rename = "signedEntryTimestamp")]
+    pub signed_entry_timestamp: String,
+}
+
+/// Convenient struct for verifying the `signedEntryTimestamp` in a Rekor LogEntry.
+///
+/// This bundle can be verified using the public key from Rekor. The public key can
+/// be obtained from the `/api/v1/log/publicKey` Rest API. For `sigstore.dev`, it is a PEM-encoded
+/// x509/PKIX public key. The PEM-encoded content can be parsed into a `p256::ecdsa::VerifyingKey`
+/// using `unmarshal_pem_to_p256_public_key`.
+pub struct RekorSignatureBundle {
+    /// Canonicalized JSON representation of a subset of a Rekor LogEntry fields that are signed
+    /// to generate `signedEntryTimestamp` (also a field in the Rekor LogEntry). These fields
+    /// include body, integratedTime, logID and logIndex.
+    pub canonicalized: Vec<u8>,
+
+    /// Signature over the canonicalized JSON document. This is obtained by decoding the
+    /// Base64-encoded `signedEntryTimestamp` in the Rekor LogEntry.
+    pub signature: p256::ecdsa::Signature,
+}
+
+/// Convertor for creating a RekorSignatureBundle from a Rekor LogEntry as described in https://github.com/sigstore/rekor/blob/4fcdcaa58fd5263560a82978d781eb64f5c5f93c/openapi.yaml#L433-L476.
+impl TryFrom<&LogEntry> for RekorSignatureBundle {
+    type Error = anyhow::Error;
+
+    fn try_from(log_entry: &LogEntry) -> anyhow::Result<Self> {
+        // Create a copy of the LogEntry, but skip the verification.
+        let entry_subset = LogEntry {
+            body: log_entry.body.clone(),
+            integrated_time: log_entry.integrated_time,
+            log_id: log_entry.log_id.clone(),
+            log_index: log_entry.log_index,
+            verification: None,
+        };
+
+        // Canonicalized JSON document that is signed.
+        let canonicalized = serde_jcs::to_string(&entry_subset)
+            .context("couldn't create canonicalized json string")?;
+        let canonicalized = canonicalized.as_bytes().to_vec();
+
+        // Extract the signature from the LogEntry.
+        let sig_base64 = log_entry
+            .verification
+            .as_ref()
+            .ok_or(anyhow::anyhow!("no verification field in the log entry"))?
+            .signed_entry_timestamp
+            .clone();
+        let sig_bytes = sig_base64.as_str().as_bytes();
+        let sig =
+            base64::decode(sig_bytes).context("couldn't decode Base64 signedEntryTimestamp")?;
+        let signature = Signature::from_der(&sig).context("invalid ASN.1 signature")?;
+
+        Ok(Self {
+            canonicalized,
+            signature,
+        })
+    }
+}
+
+/// Verifies a Rekor LogEntry, by:
+/// (1) verifying the signature in `signedEntryTimestamp` using Rekor's public key,
```

Comment:
super nit: I think it is customary to have a blank line after the first sentence of documentation.